app.js

```diff
@@ -140,7 +156,18 @@ function on_video_played() {
 
     });
 
-    setTimeout(on_player_settings_clicked, 1000);
+
+    // Check whether the settings button exists since firefox prevents it from loading due to autoplay with sound prevention
+    function loadCheckSettings() {
+        if (document.querySelector('.player-settings-icon')) {
+            on_player_settings_clicked()
+        } else {
+            setTimeout(loadCheckSettings, 25);
+        }
+    }
+    
+    loadCheckSettings()
```

Comment:
can this be moved into the run_extension_functions() 
just so that it be easy to know which functions run on when extention starts

Reply:
I've looked, it seems that moving the function call shouldn't cause problems, and the function itself probably shouldn't have been defined inside another function anyway.

I moved it.